include/libtorrent/torrent_info.hpp

```diff
@@ -501,7 +497,7 @@ namespace libtorrent
 		// ``metadata()`` returns a the raw info section of the torrent file. The size
 		// of the metadata is returned by ``metadata_size()``.
 		int metadata_size() const { return m_info_section_size; }
-		boost::shared_array<char> metadata() const
+		std::shared_ptr<char> metadata() const
```

Comment:
ok, agree that `boost::shared_array` is the best tool for now, I will revert all the changes related to shared_array, I want to easy the review and merge